test/native/helpers.js

```diff
@@ -43,19 +43,28 @@ provideToNativeHtml.mockImplementation( ( html ) => {
  * @param {import('react').ReactNode} [options.component] A specific editor component to render.
  * @return {import('@testing-library/react-native').RenderAPI} A Testing Library screen.
  */
-export function initializeEditor( props, { component = Editor } = {} ) {
+export async function initializeEditor( props, { component = Editor } = {} ) {
 	// Portions of the React Native Animation API rely upon these APIs. However,
-	// Jest's 'legacy' fake timer mutate these globals, which breaks the Animated
+	// Jest's 'legacy' fake timers mutate these globals, which breaks the Animated
 	// API. We preserve the original implementations to restore them later.
 	const originalRAF = global.requestAnimationFrame;
 	const originalCAF = global.cancelAnimationFrame;
 
-	// During editor initialization, asynchronous store resolvers rely upon
+	// During editor initialization, asynchronous store resolvers leverage
+	// `apiFetch` to fetch data from the REST API. The resolvers also rely upon
 	// `setTimeout` to run at the end of the current JavaScript block execution.
 	// In order to prevent "act" warnings triggered by updates to the React tree,
-	// we leverage fake timers to manually tick and await the resolution of the
-	// current block execution before proceeding.
+	// we manually tick fake timers and await the resolution of `apiFetch`
+	// before proceeding.
 	jest.useFakeTimers( 'legacy' );
+	const mockApiResponse = Promise.resolve();
+	jest.doMock( '@wordpress/data-controls', () => {
+		const dataControls = jest.requireActual( '@wordpress/data-controls' );
+		return {
+			...dataControls,
+			apiFetch: jest.fn( () => mockApiResponse ),
+		};
+	} );
```

Comment:
I tried to debug `apiFetch` from `@wordpress/data-controls` but I couldn't manage to identify any calls during initialization. In fact, if you comment this code block the tests pass too 🤔 , which makes me wonder if we need to mock it, wdyt?

On the other hand, I understand that API requests are already being omitted as they are mocked on the Jest setup ([reference](https://github.com/WordPress/gutenberg/blob/fa5c739fc4f38aa92587a2cb31b4b0998bb6bf81/test/native/setup.js#L45-L50)), so maybe we don't need to mock them for this purpose.

Reply:
> I tried to debug `apiFetch` from `@wordpress/data-controls` but I couldn't manage to identify any calls during initialization. In fact, if you comment this code block the tests pass too 🤔 , which makes me wonder if we need to mock it, wdyt?

You are absolutely correct. It would appear the successful run was merely a side effect of the presence of `await act (() => anyPromiseResolve);`.  The specific mock was irrelevant. Good catch. 

> On the other hand, I understand that API requests are already being omitted as they are mocked on the Jest setup ([reference](https://github.com/WordPress/gutenberg/blob/fa5c739fc4f38aa92587a2cb31b4b0998bb6bf81/test/native/setup.js#L45-L50)), so maybe we don't need to mock them for this purpose.

Agreed, this mock merely hoisted the mocking further up from `@wordpress/api-fetch` to `@wordpress/data-controls`. Your first point regarding commenting the mock out makes this subject moot, however.